### PR TITLE
Space Follow Benchmarks

### PR DESCRIPTION
**Pull request by TarekkMA**

### CODE REVIEW

File: pallets/space-follows/src/benchmarking.rs [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            RawOrigin::Signed(space_follower.clone()).into(),
+            space.id,
+        )?;
+
+        // Cleanup
+        whitelist!(space_follower);
+        Pallet::<T>::unfollow_space(
+            RawOrigin::Signed(space_follower.clone()).into(),
+            space.id,
+        )?;
```
Comment (siman) [inline pallets/space-follows/src/benchmarking.rs:45]: Why do we need to unfollow a space in the benchmark for following space?

**Review by siman — CHANGES_REQUESTED** [review]

File: pallets/space-follows/src/benchmarking.rs (hunk 1 above)
Comment (TarekkMA) [reply pallets/space-follows/src/benchmarking.rs:45]: I've tried to run benchmarks without it and got an error that I've already followed the space from the next iteration of the benchmarks.

And a fix was with this cleanup unfollow

File: pallets/space-follows/src/benchmarking.rs [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    }
```
Comment (F3Joule) [inline pallets/space-follows/src/benchmarking.rs:46]:
```suggestion
    }: _(RawOrigin::Signed(space_follower.clone()), space.id)
```

File: pallets/space-follows/src/weights.rs [hunk 3]
```diff
@@ -0,0 +1,92 @@
+//! Autogenerated weights for pallet_space_follows
```
Comment (F3Joule) [inline pallets/space-follows/src/weights.rs:1]: Good for now, but should be re-generated once automation tool is done.

**Review by F3Joule — APPROVED** [review]

File: pallets/space-follows/src/benchmarking.rs [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+    follow_space {
+        let space_owner_origin = RawOrigin::Signed(account::<T::AccountId>("SpaceOwner", 2, 0));
+        let space_follower = account::<T::AccountId>("SpaceFollower", 1, 0);
+
+        let space = create_dummy_space::<T>(space_owner_origin.clone())?;
+    }: _(RawOrigin::Signed(space_follower.clone()), space.id)
+    verify {
+        let space = pallet_spaces::SpaceById::<T>::get(space.id)
+            .ok_or(DispatchError::Other("Space not found"))?;
```
Comment (siman) [inline pallets/space-follows/src/benchmarking.rs:36]: It looks like it's unneeded. We already have access to `space`

File: pallets/space-follows/src/benchmarking.rs [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        let space_owner_origin = RawOrigin::Signed(account::<T::AccountId>("SpaceOwner", 2, 0));
+        let space_follower = account::<T::AccountId>("SpaceFollower", 1, 0);
+
+        let space = create_dummy_space::<T>(space_owner_origin.clone())?;
+        Pallet::<T>::follow_space(RawOrigin::Signed(space_follower.clone()).into(),space.id)?;
+
+    }: _(RawOrigin::Signed(space_follower.clone()), space.id)
+    verify {
+        let space = pallet_spaces::SpaceById::<T>::get(space.id)
+            .ok_or(DispatchError::Other("Space not found"))?;
```
Comment (siman) [inline pallets/space-follows/src/benchmarking.rs:53]: It looks like it's unneeded. We already have access to `space`

File: pallets/space-follows/src/benchmarking.rs [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+
+        ensure!(!SpaceFollowers::<T>::get(space.id).contains(&space_follower), "SpaceFollowers didn't update");
```
Comment (siman) [inline pallets/space-follows/src/benchmarking.rs:55]:
```suggestion
        ensure!(!SpaceFollowers::<T>::get(space.id).contains(&space_follower), "SpaceFollowers was not updated");
```

File: pallets/space-follows/src/benchmarking.rs [hunk 7]
```diff
@@ -0,0 +1,59 @@
...
+        ensure!(!SpaceFollowedByAccount::<T>::get(&(space_follower.clone(), space.id)), "SpaceFollowedByAccount didn't update");
```
Comment (siman) [inline pallets/space-follows/src/benchmarking.rs:56]:
```suggestion
        ensure!(!SpaceFollowedByAccount::<T>::get(&(space_follower.clone(), space.id)), "SpaceFollowedByAccount was not updated");
```

File: pallets/space-follows/src/benchmarking.rs [hunk 8]
```diff
@@ -0,0 +1,59 @@
...
+        ensure!(!SpacesFollowedByAccount::<T>::get(&space_follower).contains(&space.id), "SpacesFollowedByAccount didn't update");
```
Comment (siman) [inline pallets/space-follows/src/benchmarking.rs:57]:
```suggestion
        ensure!(!SpacesFollowedByAccount::<T>::get(&space_follower).contains(&space.id), "SpacesFollowedByAccount was not updated");
```

File: pallets/space-follows/src/benchmarking.rs [hunk 9]
```diff
@@ -0,0 +1,59 @@
...
+        let space_owner_origin = RawOrigin::Signed(account::<T::AccountId>("SpaceOwner", 2, 0));
+        let space_follower = account::<T::AccountId>("SpaceFollower", 1, 0);
+
+        let space = create_dummy_space::<T>(space_owner_origin.clone())?;
+    }: _(RawOrigin::Signed(space_follower.clone()), space.id)
+    verify {
+        let space = pallet_spaces::SpaceById::<T>::get(space.id)
+            .ok_or(DispatchError::Other("Space not found"))?;
+
+        ensure!(SpaceFollowers::<T>::get(space.id).contains(&space_follower), "SpaceFollowers didn't update");
```
Comment (siman) [inline pallets/space-follows/src/benchmarking.rs:38]:
```suggestion
        ensure!(SpaceFollowers::<T>::get(space.id).contains(&space_follower), "SpaceFollowers was not updated");
```

File: pallets/space-follows/src/benchmarking.rs [hunk 10]
```diff
@@ -0,0 +1,59 @@
...
+        ensure!(SpaceFollowedByAccount::<T>::get(&(space_follower.clone(), space.id)), "SpaceFollowedByAccount didn't update");
```
Comment (siman) [inline pallets/space-follows/src/benchmarking.rs:39]:
```suggestion
        ensure!(SpaceFollowedByAccount::<T>::get(&(space_follower.clone(), space.id)), "SpaceFollowedByAccount was not updated");
```

File: pallets/space-follows/src/benchmarking.rs [hunk 11]
```diff
@@ -0,0 +1,59 @@
...
+        ensure!(SpacesFollowedByAccount::<T>::get(&space_follower).contains(&space.id), "SpacesFollowedByAccount didn't update");
```
Comment (siman) [inline pallets/space-follows/src/benchmarking.rs:40]:
```suggestion
        ensure!(SpacesFollowedByAccount::<T>::get(&space_follower).contains(&space.id), "SpacesFollowedByAccount was not updated");
```

**Review by siman — CHANGES_REQUESTED** [review]

**Review by siman — APPROVED** [review]

File: pallets/space-follows/Cargo.toml [hunk 12]
```diff
@@ -12,9 +12,11 @@ categories = ['cryptography::cryptocurrencies']
 
 [features]
 default = ['std']
+runtime-benchmarks = ["frame-benchmarking"]
```
Comment (F3Joule) [inline pallets/space-follows/Cargo.toml:15]:
```suggestion
runtime-benchmarks = ["frame-benchmarking/runtime-benchmarks"]
```

**Review by F3Joule — CHANGES_REQUESTED** [review]
Also add `try-runtime` for a pallet.

**Review by F3Joule — APPROVED** [review]

File: .maintain/weight-template.hbs [hunk 13]
```diff
@@ -12,6 +12,7 @@
 #![cfg_attr(rustfmt, rustfmt_skip)]
 #![allow(unused_parens)]
 #![allow(unused_imports)]
+#![allow(non_snake_case)]
```
Comment (siman) [inline .maintain/weight-template.hbs:15]: Why?

Comment (TarekkMA) [reply .maintain/weight-template.hbs:15]: Because in weights there are functions generated with non snake case

**Pull request merged**